apisix/schema_def.lua

```diff
@@ -37,7 +37,7 @@ local id_schema = {
     }
 }
 
-local host_def_pat = "^\\*?[0-9a-zA-Z-._]+$"
+local host_def_pat = "^\\*?[0-9a-zA-Z-._\\[\\]:]+$"
```

Comment:
Can validate ipv4/ipv6 more detail ?

Reply:
I would prefer to a simple regex check as a complex check is error-prone (may reject valid input)

Reply:
IPv6 schema: https://github.com/apache/apisix/blob/master/apisix/schema_def.lua#L56

Reply:
> IPv6 schema: https://github.com/apache/apisix/blob/master/apisix/schema_def.lua#L56
> 
> can we reuse it?

Let my try

Reply:
My point that it is hard to use the same regular exprssion to valitate ipv4 and ipv6.

Reply:
> I would prefer to a simple regex check as a complex check is error-prone (may reject valid input)

So I'm leaning towards a more conservative way of fixing it, like this currently. @zhendongcmss